application-admintools-default/src/main/java/com/xwiki/admintools/internal/wikiUsage/UsageDataProvider.java

```diff
@@ -0,0 +1,213 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.wikiUsage;
+
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.activeinstalls2.internal.PingDataProvider;
+import org.xwiki.activeinstalls2.internal.data.DatabasePing;
+import org.xwiki.activeinstalls2.internal.data.ExtensionPing;
+import org.xwiki.activeinstalls2.internal.data.Ping;
+import org.xwiki.activeinstalls2.internal.data.ServletContainerPing;
+import org.xwiki.activeinstalls2.internal.data.UsersPing;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryFilter;
+import org.xwiki.query.QueryManager;
+import org.xwiki.wiki.descriptor.WikiDescriptor;
+
+import com.xwiki.admintools.WikiSizeResult;
+
+/**
+ * Retrieves {@link Ping} data from Active Installs 2.
+ *
+ * @version $Id$
+ */
+@Component(roles = UsageDataProvider.class)
+@Singleton
+public class UsageDataProvider implements Initializable
+{
+    private static final String METADATA_NAME = "name";
+
+    private static final String METADATA_VERSION = "version";
+
+    @Inject
+    @Named("database")
+    private PingDataProvider databasePingDataProvider;
+
+    @Inject
+    @Named("servlet")
+    private PingDataProvider servletPingDataProvider;
+
+    @Inject
+    @Named("extensions")
+    private PingDataProvider extensionsPingDataProvider;
+
+    @Inject
+    @Named("users")
+    private PingDataProvider usersPingDataProvider;
+
+    private Ping ping;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    @Named("count")
+    private QueryFilter countFilter;
+
+    /**
+     * Initialize {@link Ping}.
+     *
+     * @throws InitializationException
+     */
+    @Override
+    public void initialize() throws InitializationException
+    {
+        this.ping = new Ping();
```

Comment:
I used the singleton because the data provided by this class it's specific to the instance, so I'd say it's no need to use a different object for every request. But considering your comment bellow regarding the thread safety and double checking on the Active installs 2, I've seen that it has no thread safety check. I'd say removing the singleton could be best.